src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -175,6 +179,28 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             globalConfig = config.Globals;
             config.OnConfigChange("Defaults", () => nodeConfig = config.GetOrCreateNodeConfigurationForSilo(name));
 
+            // Configure DI using Startup type
+            bool usingCustomServiceProvider;
+            Action<IServiceCollection> configureServices = serviceCollection =>
+            {
+                serviceCollection.AddSingleton(this);
```

Comment:
careful that you are passing `this` (and resolving it afterwards) before this object finished constructing. This can lead to very hard to troubleshoot errors, since people might resolve it and try to use it in different ways